main.go

```diff
@@ -24,11 +24,19 @@ var (
 )
 
 func getMask(vip netip.Addr, mask int) net.IPMask {
-	if mask > 0 || mask < 33 {
-		return net.CIDRMask(mask, 32)
+	if strings.Contains(vip.String(),".") {
```

Comment:
```suggestion
	if vip.Is4() {
```
https://pkg.go.dev/net/netip#Addr.Is4